File: basics/Q24/main.go
package main

import "fmt"

func main() {
	var (
		k float64
		x float64
		n float64
	)
	fmt.Print("Enter k: ")
	fmt.Scan(&k)
	fmt.Print("Enter x: ")
	fmt.Scan(&x)
	fmt.Print("Enter n: ")
	fmt.Scan(&n)
	acceleration := (k - x) * 60 / n
	fmt.Println("acceleration is", acceleration)
}

File: basics/Q24/main.rs
use std::io;
use std::str::FromStr;

fn main() {
    let k: f64 = read_input("Enter k:");
    let x: f64 = read_input("Enter x:");
    let n: f64 = read_input("Enter n:");
    let acceleration = (k - x) * 60.0 / n;
    println!("acceleration is {}", acceleration);
}

fn read_input<T: FromStr>(prompt: &str) -> T {
    println!("{}", prompt);

    let mut input = String::new();
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");

    match input.trim().parse::<T>() {
        Ok(value) => return value,
        Err(_e) => panic!("Invalid input: please try again."),
    }
}
